scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SessionListener.java

```diff
@@ -62,4 +64,13 @@ void sessionEstablished(Handshaker handshaker, DTLSSession establishedSession)
 	 * @param peer the IP address and port of the peer the handshake has been completed with
 	 */
 	void handshakeCompleted(InetSocketAddress peer);
+
+	/**
+	 * Indicates that a handshake with a given peer has failed.
+	 * 
+	 * @param peer the IP address and port of the peer the handshake has failed
+	 * @param exception the occurred handshake exception, which caused the handshake to fail
+	 */
+	void handshakeFailed(InetSocketAddress peer, HandshakeException exception);
```

Comment:
I did this, because as far as I understood it, the handshake termination is triggered by such an Exception. But I can change the parameter type to Throwable.